src/Model/Casts/CastResolver.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Mongolid\Model\Casts;
+
+use Mongolid\Model\Casts\DateTime\DateTimeCast;
+use Mongolid\Model\Casts\DateTime\ImmutableDateTimeCast;
+use Mongolid\Model\Casts\Exceptions\InvalidCastException;
+
+class CastResolver
+{
+    private const DATE_TIME = 'datetime';
+    private const IMMUTABLE_DATE_TIME = 'immutable_datetime';
+
+    public static array $validCasts = [
+        self::DATE_TIME,
+        self::IMMUTABLE_DATE_TIME,
+    ];
+
+    public static function resolve(?string $cast): ?CastInterface
+    {
+        if (is_null($cast)) {
+            return null;
+        }
+
+        return match($cast) {
+            self::DATE_TIME => new DateTimeCast(),
+            self::IMMUTABLE_DATE_TIME => new ImmutableDateTimeCast(),
+            default => throw new InvalidCastException($cast),
+        };
+    }
+}
```

Comment:
Fica a sugestão de fazer um cache dos casters.

```suggestion
class CastResolver
{
    private const DATE_TIME = 'datetime';
    private const IMMUTABLE_DATE_TIME = 'immutable_datetime';

    /**
     * @var array<string, CastInterface>
     */
    private array $cache = [];

    public static array $validCasts = [
        self::DATE_TIME,
        self::IMMUTABLE_DATE_TIME,
    ];

    

    public static function resolve(?string $castName): ?CastInterface
    {
        if (is_null($castName)) {
            return null;
        }

        if ($cast = $this->cache[$castName]) {
            return $cast;
        }

        $cast = match($castName) {
            self::DATE_TIME => new DateTimeCast(),
            self::IMMUTABLE_DATE_TIME => new ImmutableDateTimeCast(),
            default => throw new InvalidCastException($cast),
        };
        
        $this->cache[$castName] = $cast;
        
        return $cast;
    }
}
```

Reply:
Mas qual a vantagem de fazer esse cache aqui? 

Porque basicamente ele só vai adicionar uma comparação antes da outra e retornar a instância se ela existir. O tempo de execução gasto pra isso não seria praticamente o mesmo do atual, que só cria uma nova instância e libera pra uso?

Reply:
Do jeito que esta hoje, para você setar criar uma instancia, para acessar criar outra,  se for fazer isso novamente, cria outra instancia.

Na hora de salvar no legacy, cria outra instancia.

Se tiver uma lista de entidades e pegar o valor de cada uma, vai criar um objeto para cada acesso.

Se estiver usando em batch... ai é uma instancia para cada acesso do batch, considerando que ele acesse a propriedade apenas uma vez.

Enfim, muitos motivos para cachear.

Reply:
Boa: agora sim eu visualizei o impacto na memória dos clientes que forem usar se o mongolid gerar e manter tanta instância viva. 

Implementarei.

Reply:
Feito: https://github.com/leroy-merlin-br/mongolid/pull/220/commits/4b8ee1078a89c5c14a514b8d690a4b361e269956